Kubernetes/windows/fluent/rubyKeepCertificateAlive/omsagenthelper.rb

```diff
@@ -0,0 +1,204 @@
+            @agent_guid = agent_guid #let's get this from certificate : more reliable?
+        end
+
+        # Return the certificate text as a single formatted string
+        def get_cert_server(cert_path)
+            cert_server = ""
+            cert_file_contents = File.readlines(cert_path)
+            for i in 1..(cert_file_contents.length-2) # skip first and last line in file
+                line = cert_file_contents[i]
+                cert_server.concat(line[0..-2])
+                if i < (cert_file_contents.length-2)
+                    cert_server.concat(" ")
+                end
+            end
+            return cert_server
+        end
+
+		# Generate the request body
+        def generate_request_body
+            begin
+                agentTopologyRequestHandler = AgentTopologyRequestHandler.new()
+                request_body_xml = agentTopologyRequestHandler.handle_request(@agent_guid, get_cert_server(@cert_path))
+			rescue => e
+				puts "Error when generating request body for OMS agent management service topology request: #{e.message}"
+            end
+
+            return request_body_xml
+        end
+
+        def get_user_agent
+            # We need to replace akswindowslog with a version number. (not necessary but it is cleaner and the right way to do it)
+            return "MicrosoftMonitoringAgent/akswindowslog"
+        end
+
+        def generate_request_headers()
+            headers = {}
+            req_date = Time.now.utc.strftime("%Y-%m-%dT%T.%N%:z")
+            headers[CaseSensitiveString.new("x-ms-Date")] = req_date
+            headers["User-Agent"] = get_user_agent
+            headers[CaseSensitiveString.new("Accept-Language")] = "en-US"
+            return headers
+        end
+
+        # create an HTTP object which uses HTTPS
+        def create_secure_http(uri, proxy={})
+            if proxy.empty?
+            http = Net::HTTP.new( uri.host, uri.port )
+            else
+            http = Net::HTTP.new( uri.host, uri.port,
+            proxy[:addr], proxy[:port], proxy[:user], proxy[:pass])
+            end
+            http.use_ssl = true
+            http.verify_mode = OpenSSL::SSL::VERIFY_PEER
+            http.open_timeout = 30
+            return http
+        end # create_secure_http
+
+        # Return a POST request with the specified headers, URI, and body, and an
+        #     HTTP to execute that request
+        def form_post_request_and_http(headers, uri_string, body, cert, key)
+            uri = URI.parse(uri_string)
+            req = Net::HTTP::Post.new(uri.request_uri, headers)
+            req.body = body
+            http = create_secure_http(uri)
+            http.cert = cert
+            http.key = key
+            return req, http
+        end
+
+        # Call dotnet exectuable here to request new certificate
+        def renew_certs
+            systemCommandResult = system("C:\\opt\omsagentwindows\\certgenerator\\CertificateGenerator.exe")
```

Comment:
Restart the container by failing the liveness probe since go plugin has loaded the certificates and isn't going to reload them